woocommerce/woocommerce-gutenberg-products-block
# Switch from Select to Combobox for Country and State Inputs

This PR implements a new Combobox control which essentially allows text input to search a list of values. This is useful for country and state fields and is similar to what Woo Core has on the core checkout.

The control being used is the [ComboBox Control from Gutenberg](https://developer.wordpress.org/block-editor/reference-guides/components/combobox-control/).

Fixes #1752

Looking for feedback and ideas to further polish. To get the label working I needed to simulate onBlur. Autofill was also challenging. I ended up making it search when the input changes which causes an immediate update so I am not 100% sure this is a good solution.

#### Accessibility

Tested Keyboard controls.

### Screenshots

![Screenshot 2021-06-18 at 13 50 00](https://user-images.githubusercontent.com/90977/122563859-a16fb580-d03c-11eb-9e07-2f54f4e330f7.png)
![Screenshot 2021-06-18 at 13 49 33](https://user-images.githubusercontent.com/90977/122563860-a2084c00-d03c-11eb-9f06-3ed2c4a50e9d.png)
![Screenshot 2021-06-18 at 13 48 53](https://user-images.githubusercontent.com/90977/122563863-a2a0e280-d03c-11eb-8b82-bab75d0c84ec.png)

### How to test the changes in this Pull Request:

1. Try searching for values in the country input. Value should persist on change.
2. Repeat for state input (for countries with states)
3. Test autofill populates the field
4. Test on a mobile browser (I used mobile view on Edge)
5. Test on a variety of themes

I went through all the TwentyX themes and Storefront.

### Changelog

> Checkout: Switch from select element to combobox for country and state inputs so contents are searchable
